pycbc/inference/sampler/multinest.py

```diff
@@ -396,4 +402,7 @@ def from_config(cls, cp, model, nprocesses=1, use_mpi=False):
                            v is not None}
         obj = cls(model, nlivepoints, constraints=constraints,
                   **optional_kwargs)
+        obj.setup_output(obj, output_file)
+        if not obj.new_checkpoint:
+            obj.resume_from_checkpoint()
```

Comment:
Sorry to change my mind a bit, but after a little more thought it seems a little misleading to call this method here when the method just passes. I'd prefer these lines be removed if they're not doing anything. Same comment goes for the corresponding lines in the dynesty and cpnest samplers.

Reply:
What if he sticks a logging warning message in their `resume_from_checkpoint` functions that say "checkpointing not supported yet"?

Reply:
I'm not sure about cpnest and dynesty, but for multinest it actually does support checkpointing, it's just done through its own auxiliary files.

Reply:
Oh, I didn't realize that about multinest. In that case, yeah, the `resume_from_checkpoint` call should be removed. Where are the files stored? It's a little unnerving that the run will recover from a checkpoint using files  that the executable is unaware of. For instance, is it possible to accidentally resume from the wrong file, or not resume because the files are getting written in a different place than multinest expects?

For dynesty, Sumit is going to be adding checkpointing to that next. Getting ready for that is what prompted this PR.

CPNest does something similar with saving its own files, but it only kinda sorta works, and definitely doesn't work in condor. Fixing that properly may require a fix to CPNest itself; since we haven't been using it much, that's low on the priority list. So for all intents and purposes, our implementation of CPNest doesn't support checkpointing.

Reply:
Okay, then removing the lines here but leaving them in place for dynesty and cpnest sounds good to me.

I agree that the auxiliary file situation with multinest is a little cumbersome, but I did try to mitigate some risk by forcing the sampler to use the pycbc output file as a base for all the auxiliary files so they'll always be stored in the same location. It is still possible though to run into unexpected resume behavior in a couple (hopefully rare) edge cases. For instance, if the pycbc output files exist but the auxiliary files do not, the run will start from scratch. Conversely if the pycbc output files don't exist but the auxiliary files do, the run will resume.

In the future it would be good to integrate the files somehow, either by making `pycbc_inference` aware of the auxiliary files, or by writing all the necessary resume info to the pycbc files and having the auxiliary files generated on resume.